eip-0029.md

```diff
@@ -39,8 +39,7 @@ reserved and mandatory, registers 4 to 9 are optional and can be used as input f
 or are used for token minting (see EIP-4).
 
 Registers must be densely packed. It is not possible to use register 9 without adding register 4
-to 8. However, if necessary registers of `Unit` type can be added to fill the empty slot
-with only 1 byte per register of overhead.
+to 8. However, if necessary registers can be added to fill the empty slot.
```

Comment:
```suggestion
to 8. However, if necessary registers can be added to fill the empty slot. 
This can only be done safely for boxes protected by `compatible` contracts:
- p2pk contracts are all compatible
- contracts not referring to registers of SELF box and also not referring to registers of INPUTS

The second category of contracts can be identified through analysis of ErgoTree
```